### PR TITLE
Add what are tests and testing frameworks

### PR DESCRIPTION
**Pull request by schaui6**
@akanshmurthy Hey buddy, sorry for lagging.  Let me know what you think :)

### CODE REVIEW

File: sites/en/testing/testing_frameworks.step [hunk 1]
```diff
@@ -1,7 +1,97 @@
 message <<-MARKDOWN
-  Testing frameworks.
+# What's a testing framework?
+A Testing framework is an execution environment for automated tests. Think of it as the set of assumptions that reminds you when you veer away from those assumptions. In short Test frameworks helps teams organize their test suites and in turn
```
Comment (akanshmurthy) [inline sites/en/testing/testing_frameworks.step]: Lowercase the t in Testing and the t in Test.

File: sites/en/testing/testing_frameworks.step [hunk 2]
```diff
@@ -1,7 +1,97 @@
...
+help improve the efficiency of testing. 
+
+# Types of testing frameworks
+There are many testing frameworks that work great.  Mini Test is the default testing framework in Rails 5. However, we will be using the RSpec testing framework instead.
```
Comment (akanshmurthy) [inline sites/en/testing/testing_frameworks.step:7]: Can we get a few lines on differences between MiniTest and RSpec?

File: sites/en/testing/what_are_tests.step [hunk 3]
```diff
@@ -1,7 +1,23 @@
 message <<-MARKDOWN
-  What are tests?
+# What Are Tests?
+Tests are ways of testing our code to see if they are performing the way we have intended it behave.
```
Comment (akanshmurthy) [inline sites/en/testing/what_are_tests.step:2]: to see if "it" is performing instead of "they."

File: sites/en/testing/what_are_tests.step [hunk 4]
```diff
@@ -1,7 +1,23 @@
...
+
+# Example
+ For example, lets say we wanted to create a program that models an orange tree.  Well, what defines an orange tree to us? Is it safe to say that we need at least two objects, an orange object and a tree object? Can our tree mature and bear more fruit at a certain age? Could our oranges ripen and fall from the tree?  Sounds like the list can become lengthy right? Lets stop here for a few minutes and see what we need to fulfill these user stories.
```
Comment (akanshmurthy) [inline sites/en/testing/what_are_tests.step:5]: Apostrophe missing in lets x2.

File: sites/en/testing/what_are_tests.step [hunk 5]
```diff
@@ -1,7 +1,23 @@
...
+ 
+How do we determine if an orange is ripe? Well, we can have oranges age and if it is at least 30 days old, then it will fall from the tree.  We would need to write a test to check for ripeness and a test to check if it falls at ripeness.
+
+# Now That We Have Tests
+You could think of tests as a requirement list.  Every time a change is made we want to test to see if our code still meets those requirements. For example, as our program becomes increasingly complex and we want to have our tree to have a certain lifespan, would it make sense for our tree to continue to create oranges after exceeding that lifespan?   
+
+# Why Is It Important That We Test?
+For situations like our tree's lifespan is a prime example of why tests are important.  As our program become more complex over time the tests tells us that a basic requirement is not fulfilled and needs to be addressed.  Can you imagine looking at pages and pages of code that was written by someone else or many years ago without a way to trace exactly where the bug is?  Tests can help!
```
Comment (akanshmurthy) [inline sites/en/testing/what_are_tests.step:19]: Can we mention TDD in this as well? :)

**Comment by akanshmurthy** [comment]
Can we get a screenshot or two of the changes and how they look? Added my review comments.

**Comment by schaui6** [comment]
@akanshmurthy 
What are tests:
1 ) https://github.com/schaui6/schaui6.github.io/blob/master/Screen%20Shot%202016-12-05%20at%209.51.22%20PM.png
2) https://github.com/schaui6/schaui6.github.io/blob/master/Screen%20Shot%202016-12-05%20at%209.51.27%20PM.png

Testing Frameworks:
1) https://github.com/schaui6/schaui6.github.io/blob/master/Screen%20Shot%202016-12-05%20at%209.52.33%20PM.png
2) https://github.com/schaui6/schaui6.github.io/blob/master/Screen%20Shot%202016-12-05%20at%209.52.45%20PM.png
3) https://github.com/schaui6/schaui6.github.io/blob/master/Screen%20Shot%202016-12-05%20at%209.54.04%20PM.png
4) https://github.com/schaui6/schaui6.github.io/blob/master/Screen%20Shot%202016-12-05%20at%209.53.03%20PM.png

**Review by akanshmurthy — CHANGES_REQUESTED** [review]

File: sites/en/testing/what_are_tests.step (hunk 4 above)
Comment (schaui6) [reply sites/en/testing/what_are_tests.step:5]: are you sure? lol that doesn't seem right

File: sites/en/testing/what_are_tests.step (hunk 5 above)
Comment (schaui6) [reply sites/en/testing/what_are_tests.step:19]: the thought has crossed my mind, but I worry our testathon isn't exactly tdd.  I can mention it either way

**Pull request closed without merging**